ui-macos/lokinet/DNSManager.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  DNSManager.swift
+//  lokinet
+//
+//  Copyright © 2019 Loki. All rights reserved.
+//
+
+import Foundation
+
+func split(str: String?) -> [String] {
+    let res = str?.components(separatedBy: NSCharacterSet.whitespacesAndNewlines) ?? []
+    return res.filter({!$0.isEmpty})
+}
+
+class DNSManager {
+    static let netSetup = URL(fileURLWithPath: "/usr/sbin/networksetup")
+
+    let oldDNSSettings: [String]
+    let interface: String
+
+    static func getOldSettings(interface: String) -> [String] {
+        let netprocess = Process()
+        netprocess.executableURL = DNSManager.netSetup
+        netprocess.arguments = ["-getdnsservers", interface]
+
+        do {
+            let pipe = Pipe()
+            netprocess.standardOutput = pipe
+            try netprocess.run()
+            let data = pipe.fileHandleForReading.readDataToEndOfFile()
+            let asStr = String(data: data, encoding: .ascii)
+
+            return split(str: asStr).filter({$0 != "127.0.0.1"})
+        } catch {
+            return []
+        }
+    }
+
+    func setNewSettings() throws {
+        let netprocess = Process()
+        netprocess.executableURL = DNSManager.netSetup
+
+        netprocess.arguments = ["-setdnsservers", self.interface]
+        netprocess.arguments?.append("127.0.0.1")
+        netprocess.arguments?.append(contentsOf: oldDNSSettings)
```

Comment:
the dns server should be ONLY lokinet so that leaks don't make lookups return false positives.